src/main/java/info/bluefloyd/jenkins/RESTClient.java

```diff
@@ -429,7 +450,7 @@ private RestResult doPost(URL url, String bodydata) throws IOException {
     result.setResultCode(conn.getResponseCode());
     result.setResultMessage(output.toString());
 
-    if ((conn.getResponseCode() == 200) || (conn.getResponseCode() == 201)) {
+    if ((conn.getResponseCode() == 200) || (conn.getResponseCode() == 201) || (conn.getResponseCode() == 204)) {
```

Comment:
Seems like each operation is expecting different status code, It might be easier if we completly removed this validResult property and let caller methods check the responsCode